### PR TITLE
Add ingest zip.

### PR DESCRIPTION
**Pull request by yqjiang**

### CODE REVIEW

File: includes/manage_issue.inc [hunk 1]
```diff
@@ -36,3 +36,281 @@ function islandora_newspaper_ingest_page(AbstractObject $issue) {
...
+ */
+function islandora_newspaper_add_pages($repository, $config, $pages_directory, &$context) {
+  module_load_include('inc', 'islandora', 'includes/IslandoraTuque');
+  module_load_include('inc', 'islandora_paged_content', 'includes/utilities');
+  $context['results']['pages_directory'] = $pages_directory;
+
+  // Create object.
+  $object = $repository->constructObject($config['namespace']);
+  $object->label = pathinfo($config['image'], PATHINFO_FILENAME);
+  $ds_label = pathinfo($config['image'], PATHINFO_BASENAME);
```
Comment (jordandukart) [inline includes/manage_issue.inc]: This should be made to reflect the recent changes in the book zipped ingest here: https://github.com/Islandora/islandora_solution_pack_book/blob/7.x/includes/manage_book.inc#L275-L278

File: includes/manage_issue.inc [hunk 2]
```diff
@@ -36,3 +36,281 @@ function islandora_newspaper_ingest_page(AbstractObject $issue) {
...
+
+  );
+
+  $form['issue_pid'] = array(
+    '#type' => 'hidden',
+    '#value' => $issue_pid,
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Add files to issue.'),
```
Comment (jordandukart) [inline includes/manage_issue.inc]: This shouldn't have a period on the end of the button #value.

**Comment by jordandukart** [comment]
Should this be done in paged content to encompass both book and newspaper to fall under DRY? Thoughts @adam-vessey, @jonathangreen?

**Comment by adam-vessey** [comment]
Probably should be... Really, only thing different would be the content model which gets applied, yeah? At least pull the form and batch processing out into paged content, make the form take the content model to apply as a parameter, and then can update book and newspaper to call the form appropriately.

File: includes/manage_issue.inc [hunk 3]
```diff
@@ -36,3 +36,281 @@ function islandora_newspaper_ingest_page(AbstractObject $issue) {
...
+/**
+ * Local menu action to present a zipped file ingest form.
+ *
+ * @param AbstractObject $object
+ *   The issue to ingest into.
+ *
+ * @return string
+ *   The HTML representation of the ingest page form.
+ */
+function islandora_newspaper_ingest_zipped_pages(AbstractObject $object) {
```
Comment (jordandukart) [inline includes/manage_issue.inc]: Shouldn't this be gone now as paged content has this similar to what's in book https://github.com/Islandora/islandora_solution_pack_book/pull/116/files

File: includes/manage_issue.inc [hunk 4]
```diff
@@ -36,3 +36,281 @@ function islandora_newspaper_ingest_page(AbstractObject $issue) {
...
+ *   The Drupal form definition.
+ * @param array $form_state
+ *   The Drupal form state.
+ * @param string $issue_pid
+ *   PID of issue into which pages are being ingested.
+ *
+ * @return array
+ *   Drupal form definition.
+ */
+function islandora_newspaper_zipped_upload_form(array $form, array &$form_state, $issue_pid) {
```
Comment (jordandukart) [inline includes/manage_issue.inc]: Same as above?

File: islandora_newspaper.module [hunk 5]
```diff
@@ -65,6 +65,15 @@ function islandora_newspaper_menu() {
       'file' => 'includes/manage_pages.inc',
     ),
     'islandora/object/%islandora_object/manage/issue/ingest' => $ingest_page_menu_item,
+    'islandora/object/%islandora_object/manage/issue/ingest_zipped' => array(
```
Comment (jordandukart) [inline islandora_newspaper.module:68]: This should be calling paged content now?

**Pull request merged**